src/main/java/com/ucl/imaginethisserver/Util/ExpoUtil.java

```diff
@@ -64,4 +84,121 @@ public boolean publish(String projectID, String projectName, UUID conversionID)
+        for (Map.Entry<String, String> e : headers.entrySet()) {
+            connection.setRequestProperty(e.getKey(), e.getValue());
+        }
+        // set request body
+        try (OutputStream os = connection.getOutputStream()) {
+            byte[] input = payload.getBytes(StandardCharsets.UTF_8);
+            os.write(input, 0, input.length);
+        }
+        logger.info("Sending request to " + url.toString());
+        // send request and get response
+        try {
+            BufferedReader in = new BufferedReader(new InputStreamReader(
+                    connection.getInputStream()));
+            String inputLine;
+            StringBuilder response = new StringBuilder();
+            while ((inputLine = in.readLine()) != null) {
+                response.append(inputLine);
+            }
+            in.close();
+            // close connection
+            connection.disconnect();
+            return new Gson().fromJson(response.toString(), JsonObject.class);
+        } catch (IOException e) {
+            logger.error("Error sending request.");
+            throw e;
+        }
+    }
+
+    /**
+     * This method will send an invitation email to the target user
+     * @param email
+     * @return the invitation ID
+     */
+    public String invite(String email) throws IOException {
+        // check if the env variables has been set
+        if (EXPO_USERNAME == null || EXPO_PASSWORD == null || EXPO_ORGANIZATION_ID == null) {
+            logger.error("Expo user credential hasn't been set! Please check the environment variables.");
+            return null;
+        }
+        // first login to the expo
+        try{
+            login(EXPO_USERNAME, EXPO_PASSWORD);
+        } catch (IOException e) {
+            logger.error("Error sending invitation.", e.fillInStackTrace());
+            return null;
+        }
+        // send request to expo graphql endpoint
+        String statement = ExpoGraphqlStatement.createUserInvitationForAccount(EXPO_ORGANIZATION_ID, email, "[VIEW]");
+        URL queryURL = new URL(EXPO_GRAPHQL_API);
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Type", "application/json");
+        headers.put("Accept", "application/json");
+        headers.put("expo-session", sessionString);
+        try {
+            JsonObject response = sendPostRequest(queryURL, headers, statement);
+            return response.get("data").getAsJsonObject()
+                                    .get("userInvitation").getAsJsonObject()
+                                    .get("createUserInvitationForAccount").getAsJsonObject()
+                                    .get("id").getAsString();
+        } catch (IOException e) {
+            logger.error("Error sending invitation: " + e.fillInStackTrace());
+            return null;
+        }
+    }
+
+    public boolean cancelInvitation(String invitationID) throws IOException {
+        if (sessionString.equals("")) {
```

Comment:
I'd make `cancelInvitation` consistent with `invite` function, so call here `login` as well (and within `login` check `sessionString` as is in my next comment)